packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramPaletteContext.tsx

```diff
@@ -43,6 +47,24 @@ export const DiagramPaletteContextProvider = ({ children }: DiagramPaletteContex
     setState((prevState) => ({ ...prevState, x: null, y: null, isOpened: false }));
   }, []);
 
+  const getLastToolInvoked = (toolSectionId: string) => {
+    return state.lastToolsInvoked.find(
+      (toolSectionWithDefaultTool) => toolSectionWithDefaultTool.toolSectionId === toolSectionId
+    )?.lastToolId;
+  };
+
+  const setLastToolInvoked = (toolSectionId: string, toolId: string) => {
+    const lastToolsInvoked: ToolSectionWithLastTool[] = state.lastToolsInvoked;
+    if (lastToolsInvoked.some((ts) => ts.toolSectionId === toolSectionId)) {
```

Comment:
Details, but I don't see why this variable is named `ts`, it's not a toolSection.

Reply:
Done.